src/service-worker.ts

```diff
@@ -18,63 +11,55 @@ declare const self: ServiceWorkerGlobalScope
 
 clientsClaim()
 
-// Precache all of the assets generated by your build process.
-// Their URLs are injected into the manifest variable below.
-// This variable must be present somewhere in your service worker file,
-// even if you decide not to use precaching. See https://cra.link/PWA
-precacheAndRoute(self.__WB_MANIFEST)
+// Precache the relevant assets generated by the build process.
+const manifest = self.__WB_MANIFEST.filter((entry) => {
+  const url = typeof entry === 'string' ? entry : entry.url
+  // If this is a language file, skip. They are compiled elsewhere.
+  if (url.endsWith('.po')) {
+    return false
+  }
+
+  // If this isn't a var woff2 font, skip. Modern browsers only need var fonts.
+  if (url.endsWith('.woff') || (url.endsWith('.woff2') && !url.includes('.var'))) {
+    return false
+  }
+
+  return true
+})
+precacheAndRoute(manifest)
 
 // Set up App Shell-style routing, so that all navigation requests
-// are fulfilled with your index.html shell. Learn more at
+// are fulfilled immediately with your index.html shell. See
 // https://developers.google.com/web/fundamentals/architecture/app-shell
 const fileExtensionRegexp = new RegExp('/[^/?]+\\.[^/]+$')
-registerRoute(
-  // Return false to exempt requests from being fulfilled by index.html.
-  ({ request, url }: { request: Request; url: URL }) => {
-    // If this isn't a navigation, skip.
-    if (request.mode !== 'navigate') {
-      return false
-    }
-
-    // If this is a URL that starts with /_, skip.
-    if (url.pathname.startsWith('/_')) {
-      return false
-    }
+registerRoute(({ request, url }: { request: Request; url: URL }) => {
+  // If this isn't a navigation, skip.
+  if (request.mode !== 'navigate') {
+    return false
+  }
 
-    // If this looks like a URL for a resource, because it contains
-    // a file extension, skip.
-    if (url.pathname.match(fileExtensionRegexp)) {
-      return false
-    }
+  // If this looks like a URL for a resource, skip.
+  if (url.pathname.match(fileExtensionRegexp)) {
+    return false
+  }
 
-    // Return true to signal that we want to use the handler.
-    return true
-  },
-  createHandlerBoundToURL(process.env.PUBLIC_URL + '/index.html')
-)
+  return true
+}, createHandlerBoundToURL(process.env.PUBLIC_URL + '/index.html'))
 
-// An example runtime caching route for requests that aren't handled by the
-// precache, in this case same-origin .png requests like those from in public/
+// Cache any public/ images (eg logos).
 registerRoute(
-  // Add in any other file extensions or routing criteria as needed.
   ({ url }) => url.origin === self.location.origin && url.pathname.endsWith('.png'),
```

Comment:
what about svg

Reply:
We don't serve any svg assets out of public.

Reply:
These are only icons, so they should be cached by the browser.